### PR TITLE
Refactors Reagent Tanks--Adds High Capacity Water Tank

### PR DESCRIPTION
**Pull request by Fox-McCloud**
re-upload of: https://github.com/ParadiseSS13/Paradise/pull/6326

Refactors the reagent tank dispensers a bit, so it's handled a bit cleaner, also adds in a few features related to them. Credit to /TG/ for most of this.

- Clicking on a water cooler (like in medbay) will create a paper cup and put it in your hands
- Bursting a water tank now spews water everywhere and wets the floor
- Can order high capacity water tanks from cargo for 12 points
  - High-capacity water tanks have 100,000 units of water in them 
- Regular water tanks cost reduced from 8 points to 6 points
- The amount of reagents transferred from the reagent tank to the container is based upon the container's transfer rate as opposed to the trasnfer rate of the tank (this was silly and inconsistent)

Fix
- Fixes spilling containers when putting a beaker with reagents in the ice cream machine

Admin
- Admin warning for setting someone on fire with a welder

Code Stuffs
- Reagent dispenser spawning is handled with `reagent_id` var on the container and the amount created is controlled by `tank_volume`
- Removed the coolant tank's special handling for being damaged/destroyed (I wasn't going to bother refactoring this when there's only two on the map and Xenoarch materials are slated to be axed, anyway)
- Welding fuel refilling handled in a tank's `attackby` as opposed to the welder's `afterattack`
- Ensures rigged welding fuel tanks garbage collect

Went ahead and mapped it in for the Cyberiad and Metastation. Cyberiad--I changed out the two tanks in botany and the single tank in atmos (also cleaned up the piping).  For metastation, I dropped one of of the water tanks (in the southern portion) and replaced it with a hydro tray; the two grouped together are high capacity, and the one in atmos is also high capacity. 

img

:cl: Fox McCloud
add: Water coolers now dispense cups when clicking on them with a free hand
tweak: reagent tank dispensing amount based on the container instead of the tank
fix: Fixes beaker spilling when placing a beaker into an ice cream machine
add: Adds in 100,000 unit water tanks to cargo for 12 points
tweak: regular water tank cost reduced from 8 to 6 points
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
went ahead and did the mapping.

**Comment by taukausanake** [comment]
Still don't think the tanks should be replaced in the map if you intend to keep both of them in the code. No reason to have the one with less water exist

**Comment by Fox-McCloud** [comment]
>No reason to have the one with less water exist

Course there is; the ones that are high capacity essentially serve as mobile "sinks" and are in departments that are super reliant on water (Atmos and their backpack, botany and their need to water plants). Most other places don't need this much water, and may even be motivated to use it for less than legitimate purposes (water spray spam) with such high content amounts.

**Comment by taukausanake** [comment]
A guy in a wheelchair with a water tank and fire extinguisher is going to want to do that for a long as they can. They will go for the larger tank.
It really isn't a good thing to do to just give people something better than what they have no matter how little or convenient it might be. You then have less to work forward to, no reason to upgrade because you already have the best. This is aside having a sink in Hydroponics because it is an unlimited source of water. Given that, neither water tank is needed in Hydroponics and they don't need to drag it behind them if they are staying in the same room. Even less reason when we have the hose system for the trays now

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -190,76 +195,65 @@
...
-/obj/structure/reagent_dispensers/beerkeg/New()
-	..()
-	reagents.add_reagent("beer",1000)
+	desc = "Beer is liquid bread, it's good for you..."
+	icon_state = "beer"
+	reagent_id = "beer"
 
 /obj/structure/reagent_dispensers/beerkeg/blob_act()
-	explosion(loc,0,3,5,7,10)
+	explosion(loc, 0, 3, 5, 7, 10)
```
Comment (Crazylemon64) [inline code/modules/reagents/reagent_dispenser.dm]: what did they put in that beer

**Pull request merged**